src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,149 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+import { client } from '../utils/fetchClient';
+import classNames from 'classnames';
+
+type ChangeEvent =
+  | React.ChangeEvent<HTMLInputElement>
+  | React.ChangeEvent<HTMLTextAreaElement>;
+
+type InputNames = 'name' | 'mail' | 'comment';
+
+interface Props {
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+  selectedPost: Post;
+  setLoadingError: React.Dispatch<React.SetStateAction<boolean>>;
+  setShowForm: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  setComments,
+  selectedPost,
+  setLoadingError,
+  setShowForm,
+}) => {
+  const [userName, setUserName] = useState('');
+  const [userMail, setUserMail] = useState('');
+  const [commentText, setCommentText] = useState('');
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [userNameError, setUserNameError] = useState(false);
+  const [userMailError, setUserMailError] = useState(false);
+  const [commentTextError, setCommentTextError] = useState(false);
+
+  const handleFullForm = (e: ChangeEvent, inputName: InputNames) => {
+    switch (inputName) {
+      case 'name': {
+        setUserName(e.target.value);
+        setUserNameError(false);
+        break;
+      }
+
+      case 'mail': {
+        setUserMail(e.target.value);
+        setUserMailError(false);
+        break;
+      }
+
+      case 'comment': {
+        setCommentText(e.target.value);
+        setCommentTextError(false);
+        break;
+      }
+    }
+  };
+
+  const cleanForm = () => {
+    setUserName('');
+    setUserMail('');
+    setCommentText('');
+
+    setUserNameError(false);
+    setUserMailError(false);
+    setCommentTextError(false);
+  };
+
+  const saveInputDataOnSuccess = () => {
+    setUserName(userName);
+    setUserMail(userMail);
+    setCommentText('');
+  };
+
+  const validateInput = (input: string): boolean => {
+    const doesEmpty = input.trim().length > 0;
+
+    return doesEmpty;
+  };
+
+  const addCommetns = async () => {
```

Comment:
There is a typo in the function name `addCommetns`. It should be corrected to `addComments` to reflect its purpose accurately.